app/src/main/java/com/example/task_planner_app/ui/fragments/CreateTaskFragment.kt

```diff
@@ -0,0 +1,121 @@
+package com.example.task_planner_app.ui.fragments
+
+import android.app.Dialog
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import android.widget.Spinner
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import com.example.task_planner_app.R
+import com.example.task_planner_app.databinding.FragmentCreateTaskBinding
+import com.example.task_planner_app.ui.dialog.DatePickerFragment
+import com.example.task_planner_app.utils.Common
+import com.example.task_planner_app.viewmodel.MainActivityViewModel
+import com.shashank.sony.fancytoastlib.FancyToast
+import dagger.hilt.android.AndroidEntryPoint
+import java.text.SimpleDateFormat
+import java.util.*
+
+@AndroidEntryPoint
+class CreateTaskFragment : Fragment() {
+
+    private var loadingDialog : Dialog? = null
+    private lateinit var binding: FragmentCreateTaskBinding
+    private val viewModel by viewModels<MainActivityViewModel>()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        // Inflate the layout for this fragment
+        binding = FragmentCreateTaskBinding.inflate(inflater, container, false)
+
+        val statusList = resources.getStringArray(R.array.status_options)
+        var selectedStatus = statusList.first()
+
+        val adapter = ArrayAdapter(requireContext(), android.R.layout.simple_spinner_item, statusList)
+        binding.insertStatusTask.adapter = adapter
+
+        binding.insertStatusTask.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(
+                parent: AdapterView<*>?,
+                view: View?,
+                position: Int,
+                id: Long
+            ) {
+                selectedStatus = statusList[position]
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>?) {}
+        }
+
+        binding.insertDateTask.setOnClickListener {
```

Comment:
I suggest selected instead of insert. (insert is a reserver word more related to databases)